src/test/scala/chiseltest/tests/SingleElaboration.scala

```diff
@@ -0,0 +1,50 @@
+// See LICENSE for license details.
+
+package chiseltest.tests
+
+import chisel3._
+import chiseltest._
+import chiseltest.experimental.TestOptionBuilder._
+import chiseltest.internal.VerilatorBackendAnnotation
+import firrtl.AnnotationSeq
+import org.scalatest._
+
+class SingleElaborationTests extends FlatSpec with ChiselScalatestTester with Matchers {
+
+  behavior of "elaboration should happen once"
+
+  class ThingAMaBobBundle extends Bundle {
+    val condition = Input(Bool())
+  }
+
+  class ThingAMaBob extends Module {
+    val io = IO(new ThingAMaBobBundle)
+    val counter = RegInit(0.U(2.W))
+  }
+
+  var checkOnceVerilator = 0
+  it should ("elaborate once with Verilator") in {
+    new TestBuilder(() => {
```

Comment:
Can you have the variable increment inside the Module body itself, like
```scala
test(new Module {
  checkOnceVerilator should be (0)
  checkOnceVerilator += 1
}) { c =>
  // empty test body
}
```

You can also use the `test(...).withAnnotations(...) { c => ... }` syntax as demonstrated in [VerilatorBasicTests.scala](https://github.com/ucb-bar/chisel-testers2/blob/master/src/test/scala/chiseltest/experimental/tests/VerilatorBasicTests.scala), which I think is cleaner than creating a TestBuilder object?

Reply:
test() syntax can't be used, because it doesn't take a producer of a module, but an actual module.

Reply:
> Can you have the variable increment inside the Module body itself, like

That breaks the test...

Reply:
Wait, why does it break the test? `test(...)` is call-by-name on the DUT argument (essentially it's the same as `() => ...` but without the explicit syntax), and also I don't think it's allowed to (should assert out) elaborate a Chisel module outside a special context (that gets initialized inside somewhere `test(...)`)